node/flatpak_node_generator/providers/yarn.py

```diff
@@ -37,88 +39,72 @@ def is_git_version(version: str) -> bool:
             return len([p for p in url.path.split('/') if p]) == 2
         return False
 
+    def parse_lockfile_v1(self, lockfile: Path) -> Dict[str, Any]:
+        def _iter_lines() -> Iterator[Tuple[int, str]]:
+            indent = '  '
+            for line in lockfile.open('r'):
+                level = 0
+                while line.startswith(indent):
+                    level += 1
+                    line = line[len(indent) :]
+                yield level, line.strip()
+
+        root_entry: Dict[str, Any] = OrderedDict()
+        parent_entries = [root_entry]
+
+        for level, line in _iter_lines():
+            if line.startswith('#') or not line:
+                continue
+            assert level <= len(parent_entries) - 1
+            parent_entries = parent_entries[: level + 1]
+            if line.endswith(':'):
+                child_entry = parent_entries[-1][line[:-1]] = OrderedDict()
```

Comment:
super nit: would be nice if `line[:-1]` were saved into a `key` variable first, for clarity.